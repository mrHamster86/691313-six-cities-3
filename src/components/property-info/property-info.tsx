import * as React from 'react';

interface Props {
  rating: number;
  type: string;
  bedrooms: number;
  maxAdults: number;
  price: number;
  goods: string[];
  isPro: boolean;
  avatar: string;
  name: string;
  description: string;
}

export const PropertyInfo: React.FC<Props> = (props: Props) => {
  const MAX_RATING = 5;
  const {
    rating,
    type,
    bedrooms,
    maxAdults,
    price,
    goods,
    isPro,
    avatar,
    name,
    description,
  } = props;
  return (
    <>
      <div className="property__rating rating">
        <div className="property__stars rating__stars">
          <span
            style={{width: `${(rating / MAX_RATING) * 100}%`}}/>
          <span className="visually-hidden">Rating</span>
        </div>
        <span
          className="property__rating-value rating__value">{rating}</span>
      </div>

      <ul className="property__features">
        <li className="property__feature property__feature--entire">
          {type}
        </li>
        <li className="property__feature property__feature--bedrooms">
          {`${bedrooms} Bedrooms`}
        </li>
        <li className="property__feature property__feature--adults">
          {`Max ${maxAdults} adults`}
        </li>
      </ul>

      <div className="property__price">
        <b className="property__price-value">&euro;{price}</b>
        <span className="property__price-text">&nbsp;night</span>
      </div>

      <div className="property__inside">
        <h2 className="property__inside-title">What&apos;s inside</h2>
        <ul className="property__inside-list">
          {goods.map((good, index) => (
            <li key={index} className="property__inside-item">
              {good}
            </li>
          ))}
        </ul>
      </div>

      <div className="property__host">
        <h2 className="property__host-title">Meet the host</h2>
        <div className="property__host-user user">
          <div
            className={`property__avatar-wrapper ${isPro
              ? `property__avatar-wrapper--pro`
              : ``} user__avatar-wrapper`}>
            <img
              className="property__avatar user__avatar"
              src={`/${avatar}`}
              width="74"
              height="74"
              alt={name}
            />
          </div>
          <span
            className="property__user-name">{name}</span>
        </div>
        <div className="property__description">
          <p className="property__text">{description}</p>
        </div>
      </div>
    </>
  );
};

export default React.memo(PropertyInfo);
