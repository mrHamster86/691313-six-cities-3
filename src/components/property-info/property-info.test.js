import React from 'react';
import renderer from 'react-test-renderer';
import {PropertyInfo} from './property-info';

const rating = 1;
const type = `hotel`;
const bedrooms = 1;
const maxAdults = 1;
const price = 100;
const goods = [
  `Baby seat`,
  `Air conditioning`,
  `Towels`,
  `Washer`,
  `Laptop friendly workspace`,
  `Breakfast`,
];
const isPro = true;
const avatar = `img/avatar-angelina.jpg`;
const name = `Angelina`;
const description = `This is a place for dreamers to reset, reflect, and create.`;

it(`PropertyInfo components renders`, () => {
  const tree = renderer.create(
      <PropertyInfo
        rating={rating}
        type={type}
        bedrooms={bedrooms}
        maxAdults={maxAdults}
        price={price}
        goods={goods}
        isPro={isPro}
        avatar={avatar}
        name={name}
        description={description}
      />).toJSON();
  expect(tree).toMatchSnapshot();
});
