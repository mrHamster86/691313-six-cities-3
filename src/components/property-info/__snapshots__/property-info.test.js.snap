// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInfo components renders 1`] = `
Array [
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "20%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      1
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      hotel
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      1 Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 1 adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      100
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Baby seat
      </li>
      <li
        className="property__inside-item"
      >
        Air conditioning
      </li>
      <li
        className="property__inside-item"
      >
        Towels
      </li>
      <li
        className="property__inside-item"
      >
        Washer
      </li>
      <li
        className="property__inside-item"
      >
        Laptop friendly workspace
      </li>
      <li
        className="property__inside-item"
      >
        Breakfast
      </li>
    </ul>
  </div>,
  <div
    className="property__host"
  >
    <h2
      className="property__host-title"
    >
      Meet the host
    </h2>
    <div
      className="property__host-user user"
    >
      <div
        className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
      >
        <img
          alt="Angelina"
          className="property__avatar user__avatar"
          height="74"
          src="/img/avatar-angelina.jpg"
          width="74"
        />
      </div>
      <span
        className="property__user-name"
      >
        Angelina
      </span>
    </div>
    <div
      className="property__description"
    >
      <p
        className="property__text"
      >
        This is a place for dreamers to reset, reflect, and create.
      </p>
    </div>
  </div>,
]
`;
